pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader.go

```diff
@@ -557,35 +643,45 @@ func convertBatchGetLatestValuesRequestFromProto(pbRequest *pb.BatchGetLatestVal
 	}
 
 	request := make(types.BatchGetLatestValuesRequest)
-	for pbContractName, pbContractBatch := range pbRequest.Requests {
-		if _, ok := request[pbContractName]; !ok {
-			request[pbContractName] = []types.BatchRead{}
+	for _, batch := range pbRequest.Requests {
+		binding := convertBoundContractFromProto(batch.Contract)
+
+		if _, ok := request[binding]; !ok {
+			request[binding] = make([]types.BatchRead, len(batch.Reads))
 		}
 
-		for _, pbCall := range pbContractBatch.Reads {
-			call := types.BatchRead{ReadName: pbCall.ReadName}
-			params, err := getContractEncodedType(pbContractName, pbCall.ReadName, impl, true)
+		for idx, readCall := range batch.Reads {
```

Comment:
I find these conversion funcs easier to read if pb is prepended to protobuf types